src/launcher/services.ts

```diff
@@ -0,0 +1,40 @@
+import { launch as rootLaunch } from '../main';
+import { ServiceConfig } from '../transport/wire';
+import { fetchJson } from '../util/http';
+
+const appDirectoryHost = process.env.appDirectoryHost || 'https://app-directory.openfin.co';
+
+export interface AppDirectoryEntry {
+    manifest_url: string;
+}
+
+export const lookupServiceUrl = async (serviceName: string) => {
+    if (appDirectoryHost.length > 0) {
+        const appUrl = `${appDirectoryHost}/api/v1/apps/${serviceName}`;
+        const mani = await fetchJson(appUrl);
+        return mani.manifest_url;
+    }
+    console.error(`error getting startup url for ${serviceName}, host: ${appDirectoryHost}`);
+    return '';
```

Comment:
:neckbeard: in this case do we really want to proceed with `''` if we are calling this an error, or should we just throw to abort that service launch?